content/algorithms/assortativity/correlation.md

```diff
@@ -0,0 +1,228 @@
+    nx.draw_networkx_edges(g, pos=pos, ax=axes[i//2][i%2],edge_color='0.7')
+    axes[i//2][i%2].set_title(f"Attribute assortativity coefficient = {cr:.3}\nNumeric assortativity coefficient = {nr:.3}\nr(in,out) = {r_in_out:.3}", size=15)
+
+fig.tight_layout()
+```
+
+Nodes are colored by the `cluster` property and labeled by `num_prop` property.
+We can observe that the initial network on left side is completely assortative
+and its compliment on right side is completely disassortative.
+As we add edges between nodes of different (similar) attributes in the assortative
+(disassortative) network, the network tends to a non-assortative network and
+value of both the assortaivity coefficients tends to $0$.
+
++++
+
+The parameter `nodes` in `attribute_assortativity_coefficient` and
+`numeric_assortativity_coefficient` specifies the nodes who's edges are to be
+considered in the mixing matrix calculation.
+That is to say, if $(u,v)$ is a directed edge then the edge $(u,v)$ will be
+used in mixing matrix calculation if $u$ is in `nodes` and for undirected case
+its considered if atleast one of the $u,v$ in in `nodes`.
+
+Whereas the parameter `nodes` in `degree_assortativity_coefficient` and
+`degree_pearson_correlation_coefficient` specifies the nodes whose subgraph's
+edges are considered in the mixing matrix calculation.
+
+```{code-cell} ipython3
+nodes_list = [None,
+              [str(i) for i in range(3)],
+              [str(i) for i in range(4)],
+              [str(i) for i in range(5)],
+              [str(i) for i in range(4,8)],
+              [str(i) for i in range(5,10)]]
+fig, axes = plt.subplots(3, 2, figsize=(20, 16))
+
+def color_node(u, nodes):
+    """Utility function to give the color of a node based on its attribute """
+    if u not in nodes:
+        return '0.85'
+    if G.nodes[u]['cluster'] == 'K5':
+        return '#b3d9ff'
+    else:
+        return '#ffad99'
+
+G.add_edge('4','5')
+
+for nodes, ax in zip(nodes_list, axes.ravel()):
+    cr = nx.attribute_assortativity_coefficient(G, 'cluster',nodes=nodes)
+    nr = nx.numeric_assortativity_coefficient(G, 'num_prop',nodes=nodes)
+    ax.set_title(f"Attribute assortativity coefficient: {cr:.3}\nNumeric assortativity coefficient: {nr:.3}\nNodes = {nodes}",size=15)
+
+    if nodes is None:
+        nodes = [u for u in G.nodes()]
+    node_colors = [ color_node(u, nodes) for u in G.nodes ]
+    nx.draw_networkx_nodes(G,pos=pos, node_size=450,ax=ax,node_color = node_colors)
+    nx.draw_networkx_labels(G,pos,labels={u:u for u in G.nodes},font_size=15,ax=ax)
+    nx.draw_networkx_edges(G,pos=pos,edgelist=[ (u,v) for u,v in G.edges if u in nodes ],ax=ax,edge_color='0.3')
+fig.tight_layout()
+```
+
+In the above plots only the nodes which are considred are colored and rest are
+grayed out and only the edges which are considerd in the assortaivty calculation
+are drawn.
+
++++
+
+[^1] M. E. J. Newman, Mixing patterns in networks https://doi.org/10.1103/PhysRevE.67.026126
```

Comment:
They don't seems to be properly referenced at https://53-40210422-gh.circle-artifacts.com/0/site/_build/html/content/algorithms/assortativity/correlation.html